include/s3select.h

```diff
@@ -1218,14 +1244,27 @@ void push_like_predicate::builder(s3select* self, const char* a, const char* b)
   base_statement* expr = self->getAction()->exprQ.back();
   self->getAction()->exprQ.pop_back();
 
-  if (!dynamic_cast<variable*>(expr))
-  {
-    throw base_s3select_exception("like expression must be a constant string", base_s3select_exception::s3select_exp_en_t::FATAL);
-  }
-  else if( dynamic_cast<variable*>(expr)->m_var_type != variable::var_t::COL_VALUE)
```

Comment:
(!dynamic_cast<variable*>(expr)) 
&&
( dynamic_cast<variable*>(expr)->m_var_type == variable::var_t::COL_VALUE)
is correct approach, this condition means expression is constant expression